packages/legacy-compat/src/builders/find-record.ts

```diff
@@ -0,0 +1,125 @@
+/**
+ * @module @ember-data/legacy-compat/builders
+ */
+import { assert } from '@ember/debug';
+
+import type { StoreRequestInput } from '@ember-data/store';
+import { constructResource, ensureStringId } from '@ember-data/store/-private';
+import type { BaseFinderOptions, FindRecordOptions } from '@ember-data/store/-types/q/store';
+import type { TypeFromInstance } from '@warp-drive/core-types/record';
+import { SkipCache } from '@warp-drive/core-types/request';
+import type { ResourceIdentifierObject } from '@warp-drive/core-types/spec/raw';
+
+import { isMaybeIdentifier, normalizeModelName } from './utils';
+
+export type FindRecordRequestInput = StoreRequestInput & {
+  op: 'findRecord';
+  data: {
+    record: ResourceIdentifierObject;
+    options: FindRecordBuilderOptions;
+  };
+};
+
+export type FindRecordBuilderOptions = Omit<FindRecordOptions, 'preload'>;
+
+/**
+  This function builds a request config to find the record for a given identifier or type and id combination.
+  When passed to `store.request`, this config will result in the same behavior as a `store.findRecord` request.
+  Additionally, it takes the same options as `store.findRecord`, with the exception of `preload` (which is unsupported).
+
+  **Example 1**
+
+  ```app/routes/post.js
+  import Route from '@ember/routing/route';
+  import { findRecord } from '@ember-data/legacy-compat/builders';
+
+  export default class PostRoute extends Route {
+    model({ post_id }) {
+      return this.store.request(findRecord('post', post_id));
```

Comment:
```suggestion
      return (await this.store.request(findRecord('post', post_id))).content;
```

or similar to pull content off